### ISSUE TITLE
Improve clarity of --debug for unsupported diffs

### ISSUE THREAD

**Issue by f3rn4nd0-c354r** [skeema/skeema#52]
As you, I find it peculiar that this kind of tools are not as routine as VCS... so **congrats** on your initiative! This is a blessing to anyone that once had to manage hundreds of "ALTER TABLE" files...

I had some trouble similar to #50 and used `--debug` to understand and fix some differences in the DB. However it took me sometime to realize the actual `ON {DELETE|UPDATE}` were in fact in the repo files and not in the DB. Is this an expected behaviour?

Example:
```
2018-12-08 21:57:22 [WARN]  Skipping table table_1: unable to generate ALTER TABLE due to use of unsupported features. Use --debug for more information.
2018-12-08 21:57:22 [DEBUG] --- Expected
2018-12-08 21:57:22 [DEBUG] +++ MySQL-actual
2018-12-08 21:57:22 [DEBUG] @@ -11 +11 @@
2018-12-08 21:57:22 [DEBUG] -  CONSTRAINT `s` FOREIGN KEY (`thing_IS`) REFERENCES `a_group` (`id`)
2018-12-08 21:57:22 [DEBUG] +  CONSTRAINT `s` FOREIGN KEY (`thing_IS`) REFERENCES `a_group` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION
```

**What I was expecting:**
> -- Expected 

I thought this notation would represent what I had in my sql files/repo (my expectation).

> +++ MySQL-actual

I thought this would be the "actual"  live, running MySQL database.

**The reality:**

In fact, I found it was the other way around. ```CONSTRAINT `s` FOREIGN KEY (`thing_IS`) REFERENCES `a_group` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION``` signalled with `+` is not the actual MySQL but my sql file/repo representation.

**Comment by evanelias** [comment]
I agree that the output of unsupported features via `--debug` could be improved. It's somewhat low-priority though. Please treat `--debug` output the same way you would treat a MySQL-server stack trace: it's good information to include in an issue report, as it's intended to help the software developers debug an issue, rather than being geared to an end-user.

For background, the debug logging is not necessarily referring exactly to your sql files/repo, nor to your live DB. A table may be using unsupported features only in your files/repo (e.g. adding unsupported feature to a table that did not previously have one), only in your db (e.g. removing an unsupported feature from a table that previously had one), or in both (e.g. changing some other supported aspect of a table that has an unsupported feature). In any of these cases, Skeema diff/push will generate the "unsupported diff" message and skip the table.

Skeema does not rely on text parsing or SQL parsing to diff tables, as this would be too risky, especially across different versions/vendors of MySQL: what if Skeema's text/SQL parser had subtle differences from MySQL's? So instead it introspects `information_schema` and also uses various `SHOW` commands to build an internal representation of tables. It then verifies its understanding of the table by comparing the output of `SHOW CREATE TABLE` to what Skeema thinks the canonical `CREATE TABLE` statement should be. If they do not match, it means the table is using some feature that Skeema does not correctly understand/support, so it skips diff generation for this table as a safety precaution.

So in the output of `--debug`, the `--- Expected` section is what Skeema thinks the canonical `CREATE TABLE` should be, and `+++ MySQL-actual` section is what `SHOW CREATE TABLE` actually returns in MySQL. But I agree this could be improved to indicate whether the problem was detected on the file/repo side, the db side, or both. So I'll keep this open as a feature request. Thank you for submitting!

**Comment by evanelias** [comment]
Commit 677d8de finally improves this messaging: it now makes it clear if the problematic table was on the "from" side (original state in the live database), "to" side (desired state from the filesystem/repo), or in the attempt to compute the diff between them.

This will be included in Skeema v1.8.1, likely release target is sometime next week but may depend on when MySQL 8.0.30 comes out.
